bytecomp/translattribute.ml

```diff
@@ -80,60 +95,52 @@ let parse_inline_attribute attr =
       | _ ->
         Location.prerr_warning loc (warning txt);
         Default_inline
-    end else begin
-      (* the 'inline' and 'inlined' attributes can be used as
-         [@inline], [@inline never] or [@inline always].
-         [@inline] is equivalent to [@inline always] *)
-      let warning txt =
-        Warnings.Attribute_payload
-          (txt, "It must be either empty, 'static', 'always' or 'never'")
-      in
-      match payload with
-      | PStr [] -> Always_inline
-      | PStr [{pstr_desc = Pstr_eval ({pexp_desc},[])}] -> begin
-          match pexp_desc with
-          | Pexp_ident { txt = Longident.Lident "never" } ->
-            Never_inline
-          | Pexp_ident { txt = Longident.Lident "always" } ->
-            Always_inline
-          | Pexp_ident { txt = Longident.Lident "static" } ->
-            Static
-          | _ ->
-            Location.prerr_warning loc (warning txt);
-            Default_inline
-        end
+    end else
+      begin match get_id_payload payload with
+      | Some "never" ->
+          Never_inline
+      | Some ("always" | "") ->
+          Always_inline
       | _ ->
-        Location.prerr_warning loc (warning txt);
-        Default_inline
-    end
+          Location.prerr_warning loc
+            (Warnings.Attribute_payload
+               (txt, "It must be either empty, 'static', 'always' or 'never'"));
```

Comment:
'static' here is a leftover from the previous patch?

Reply:
P.S.: Having an auxiliary function `(string * 'a) list -> 'a option -> ('a, warning) result` would avoid this error and the maintenance issues by building the error message from the list of options.

Reply:
Ok, done.